app/dot/cpp/repository/repository/BaseRepository.java

```diff
@@ -269,4 +270,8 @@ protected Aggregation<T> getHistoryAggregation(Filter filter) {
                 .getCollection(getEntityType().getSimpleName() + "_history", getEntityType()));
     return filter != null ? historyAggregation.match(filter) : historyAggregation;
   }
+
+  public void emptyCollection() {
+    morphia.datastore().getCollection(getEntityType()).drop();
```

Comment:
Wait a second, you're not recreating the collection with its validation schema after dropping it, this is a major issue.